RecognizeDtmf/RecognizeDtmf.cs

```diff
@@ -0,0 +1,334 @@
+            }
+        }
+
+        private async Task<CallConnection> CreateCallAsync(string targetPhoneNumber)
+        {
+            try
+            {
+                //Preparting request data
+                CallSource source = new CallSource(new CommunicationUserIdentifier(callConfiguration.SourceIdentity));
+                source.CallerId = new PhoneNumberIdentifier(callConfiguration.SourcePhoneNumber);
+                var target = new PhoneNumberIdentifier(targetPhoneNumber);
+
+                var createCallOption = new CreateCallOptions(source,
+                    new List<CommunicationIdentifier>() { target },
+                    new Uri(callConfiguration.AppCallbackUrl));
+
+                Logger.LogMessage(Logger.MessageType.INFORMATION, "Performing CreateCall operation");
+
+                var call = await callClient.CreateCallAsync(createCallOption).ConfigureAwait(false);
+
+                Logger.LogMessage(Logger.MessageType.INFORMATION, $"CreateCallConnectionAsync response --> {call.GetRawResponse()}, Call Connection Id: {call.Value.CallConnectionProperties.CallConnectionId}");
+
+                Logger.LogMessage(Logger.MessageType.INFORMATION, $"Call initiated with Call Connection id: {call.Value.CallConnectionProperties.CallConnectionId}");
+
+                RegisterToCallStateChangeEvent(call.Value.CallConnectionProperties.CallConnectionId);
+
+                //Wait for operation to complete
+                await callEstablishedTask.Task.ConfigureAwait(false);
+
+                return call.Value.CallConnection;
+            }
+            catch (Exception ex)
+            {
+                Logger.LogMessage(Logger.MessageType.ERROR, string.Format("Failure occured while creating/establishing the call. Exception: {0}", ex.Message));
+                throw ex;
+            }
+        }
+
+        private async Task StartRecognizingDtmf(string targetPhoneNumber)
+        {
+            if (reportCancellationToken.IsCancellationRequested)
+            {
+                Logger.LogMessage(Logger.MessageType.INFORMATION, "Cancellation request, PlayAudio will not be performed");
+                return;
+            }
+
+            try
+            {
+                string audioFilePath = callConfiguration.AudioFileUrl + callConfiguration.AudioFileName;
+                PlaySource audioFileUri = new FileSource(new Uri(audioFilePath));
+
+                //Start recognizing Dtmf Tone
+                var recognizeOptions = new CallMediaRecognizeDtmfOptions(new PhoneNumberIdentifier(targetPhoneNumber), 1)
+                {
+                    InitialSilenceTimeout = TimeSpan.FromSeconds(15),
+                    InterToneTimeout = TimeSpan.FromSeconds(5),
+                    InterruptPrompt = true,
+                    InterruptCallMediaOperation = true,
+                    Prompt = audioFileUri,
+                    StopTones = new List<DtmfTone> { DtmfTone.Pound }
```

Comment:
Lets remove this now since it does not matter as we only accept 1, 2 and 3 tone for app scenario. And any other key is invalid and we only accept 1 tone

Reply:
ok